controllers/secret_controller.go

```diff
@@ -171,8 +170,10 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 			return reconcile.Result{}, errors.Wrapf(err, "error getting node list")
 		}
 		expectedPubKeyAnno := nodeconfig.CreatePubKeyHashAnnotation(keySigner.PublicKey())
-		escapedPubKeyAnnotation := strings.Replace(nodeconfig.PubKeyHashAnnotation, "/", "~1", -1)
-		patchData := fmt.Sprintf(`[{"op":"add","path":"/metadata/annotations/%s","value":""}]`, escapedPubKeyAnnotation)
```

Comment:
Please make the introduction of `GenerateAddPatch` a separate commit.

Replace the `Update()` with a `Patch()` should be a follow on commit.